backend/django/core/utils/util.py

```diff
@@ -231,59 +231,26 @@ def create_metadata_objects_from_csv(df, project):
             )
 
 
-def create_label_similarity_results(project):
-    """Insert data label similarity results into database using bulk_create."""
+def generate_label_embeddings(project):
+    """Create embeddings for each description of label."""
 
-    project_data = Data.objects.filter(project=project, similarityPair__isnull=True)
     project_labels = Label.objects.filter(project=project)
-
-    # get the data in dataframe form
-    data_df = pd.DataFrame(list(project_data.values("pk", "text"))).rename(
-        columns={"pk": "data_id"}
+    project_labels_descriptions = list(
+        project_labels.values_list("description", flat=True)
     )
-    res_list = []
-    for label in project_labels:
-        # set up the label series
-        label_string = f"{label.name} {label.description}"
-        label_series = pd.Series(len(data_df) * [label_string])
-
-        # compute the similarity score
-        data_df["similarity_score"] = compute_pairwise_similarities(
-            label_series, data_df["text"]
-        )
-        data_df["label_id"] = label.pk
-        res_list += data_df[["data_id", "label_id", "similarity_score"]].to_dict(
-            "records"
-        )
 
-    # compare all data with all labels at once
-    res_df = pd.DataFrame(res_list).sort_values(
-        by=["data_id", "similarity_score"], ascending=False
-    )
-    res_df["sim_rank"] = res_df.groupby("data_id")["similarity_score"].rank(
-        method="first", ascending=False
+    embeddings_request = requests.post(
+        "http://backend:8000/api/embeddings",
```

Comment:
So I'm a bit confused with this one, but I might just not have run into this use case before. Why did you have to post to an API here instead of importing `embeddings_calculations` in python and calling the function directly? Aren't you already in the backend?

Reply:
Yea, I was too at first, but it's the way Peter was showing me he has done this in the past and recommends. I think you definitely can do it there in python, but having it as part of the API makes it both 1) easier to visualize/read the embeddings API code later since all of the logic happens in the same place and 2) gives you flexibility in the future in case for some reason you do end up wanting to add embeddings from outside the python backend

Reply:
Hmm... so I don't disagree in having this capability available through the API. However, I'm not sure we're currently gaining much for point 1) especially since passing the labels through the api adds traffic. 

I don't think it would be that much more complicated to just move the `embeddings_model` declaration into a util files and the `embeddings_model.encode(request.data["strings"])` call into a function there and call it from your two api functions and `generate_label_embeddings` function directly. 

If we're worried about the embeddings code getting too complicated or difficult to find, we can always make new api_embeddings and util_embeddings files to store those api calls and their helper functions. 

Unless I'm missing something?

Reply:
Resolving this, although I'd like a comment to be added to explain what we found